src/unstract/sdk/index.py

```diff
@@ -246,9 +250,29 @@ def index(
                     x2text = X2Text(
                         tool=self.tool, adapter_instance_id=x2text_instance_id
                     )
-                    extracted_text = x2text.process(
-                        input_file_path=file_path, output_file_path=output_file_path
-                    )
+                    if enable_highlight and isinstance(
+                        x2text._x2text_instance, LLMWhisperer
+                    ):
+                        process_response: TextExtractionResult = x2text.process(
+                            input_file_path=file_path,
+                            output_file_path=output_file_path,
+                            enable_highlight=enable_highlight,
+                        )
+                        whisper_hash_value = (
+                            process_response.extraction_metadata.whisper_hash
+                        )
+
+                        metadata = {X2TextConstants.WHISPER_HASH: whisper_hash_value}
+
+                        self.tool.update_exec_metadata(metadata)
```

Comment:
@johnyrahul this would ideally write to `METADATA.json` in the path that's configured for the `Tool` (SDK's base class construct). These were part of the workflow execution related changes and since we're now calling it in the context of `index()` which can be done by `backend` as well - did you check
1.  How we generate / manage this file?
1. Does it overwrite existing contents?
1. I suggest wrapping it within a `extractor_metadata` similar to what we did for `unstract-adapters` to support any such values across extractors generically

Can you update the JIRA with its contents when it runs from prompt studio VS when it runs through a workflow?

Reply:
It was working in the prompt studio aspect as well. 
Regarding point 2 It will overwrite the existing keys if passed. Here  it is all together new key `whisper-hash`

Reply:
Also, the enable highlight argument is defaulted to a false hence this piece of code won't get executed in the prompt studio aspect.

Reply:
Got it - my only other suggestion would be to not keep this key directly as `whisper_hash` in the root of the JSON and instead follow the same structure that we've adopted for the `ExtractionResult` dataclass you had created in adapters.
Later on if we add more fields for LLMwhisperer or support new fields for other extractors we can easily enhance this